buildSrc/src/main/groovy/org/elasticsearch/gradle/NoticeTask.groovy

```diff
@@ -37,34 +37,36 @@ public class NoticeTask extends DefaultTask {
     @OutputFile
     File outputFile = new File(project.buildDir, "notices/${name}/NOTICE.txt")
 
-    /** Configurations to inspect dependencies*/
-    private List<Project> dependencies = new ArrayList<>()
+    /** Directories to include notices from */
+    private List<File> licensesDirs = new ArrayList<>()
 
     public NoticeTask() {
         description = 'Create a notice file from dependencies'
     }
 
-    /** Add notices from licenses found in the given project. */
-    public void dependencies(Project project) {
-        dependencies.add(project)
+    /** Add notices from the specified directory. */
+    public void licensesDir(File licensesDir) {
+        licensesDirs.add(licensesDir)
     }
 
     @TaskAction
     public void generateNotice() {
         StringBuilder output = new StringBuilder()
         output.append(inputFile.getText('UTF-8'))
         output.append('\n\n')
-        Set<String> seen = new HashSet<>()
-        for (Project dep : dependencies) {
-            File licensesDir = new File(dep.projectDir, 'licenses')
-            if (licensesDir.exists() == false) continue
-            licensesDir.eachFileMatch({ it ==~ /.*-NOTICE\.txt/ && seen.contains(it) == false}) { File file ->
-                String name = file.name.substring(0, file.name.length() - '-NOTICE.txt'.length())
-                appendFile(file, name, 'NOTICE', output)
-                appendFile(new File(file.parentFile, "${name}-LICENSE.txt"), name, 'LICENSE', output)
-                seen.add(file.name)
+        // This is a map rather than a set so that the sort order is the 3rd
+        // party component names, unaffected by the full path to the various files
+        Map<String, File> seen = new TreeMap<>()
```

Comment:
We don't need a map to get sorted order. Just store these in a list, and sort before printing out.